.github/workflows/integration.yml

```diff
@@ -0,0 +1,106 @@
+name: integration tests
+
+on:
+  workflow_dispatch:
+  workflow_call:
+  pull_request:
+    branches:
+      - develop
+      - releases/**
+
+concurrency:
+  group: intergrations-${{github.ref}}-${{github.event.pull_request.number || github.run_number}}
+  cancel-in-progress: true
+
+jobs:
+  integration-nix:
+    name: Integration-Build-${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-15-intel, macos-latest]
+      fail-fast: true
+    steps:
+      - name: Fetch Spack
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
+      - name: Fetch Builtin repo
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
+        with:
+            fetch-depth: 1
+            repository: spack/spack-packages
+            path: spackages
+      - name: Add builtin with shallow clone
+        run: |
+          chmod +x bin/spack
+          . share/spack/setup-env.sh
+          spack repo add --scope=spack ${{ github.workspace }}/spackages/repos/spack_repo/builtin
+      - name: "Add Integration test repo"
+        run: |
+          . share/spack/setup-env.sh
+          spack repo add ${{ github.workspace }}/var/spack/test_repos/spack_repo/integration_test
+      - name: "Spack install"
+        id: install
+        run: |
+          pip install --upgrade pip clingo
+          . share/spack/setup-env.sh
+          spack env activate ${{ github.workspace }}/var/spack/test_repos/spack_repo/integration_test/integration_env
+          spack compiler find
+          spack external find cmake ninja
+          spack concretize -U
+          spack -dv install --test root
+      - name: "Determine failure stage"
+        if: always() && (steps.install.outcome == 'failure')
+        run: |
+          . share/spack/setup-env.sh
+          echo "FAILURE_STAGE=$(spack location -S)" >> $GITHUB_ENV
+      - name: "Collect failure artifacts"
+        if: always() && (steps.install.outcome == 'failure')
+        uses: actions/upload-artifact@v4
+        with:
+          name: "broken-${{ matrix.os }}-stage"
+          path: $FAILURE_STAGE
+
+  integration-win:
```

Comment:
The only difference between this and the above is shell setup. Feels like it should be possible to combine this by making setup-env part of the shell on linux/macos?

That avoids that the two diverge, which they already do regarding `fetch-depth`.

Reply:
You can do a step `run: pip install clingo` with `if: ${{ matrix.os }} == ...` if that's the only difference